libbeat/scripts/generate_template.py

```diff
@@ -198,6 +198,22 @@ def fill_field_properties(field, defaults, path):
             properties[field.get("name")]["properties"] = prop
 
         dynamic_templates.extend(dynamic)
+    elif field.get("type") == "nested":
+        if len(path) > 0:
+            path = path + "." + field["name"]
+        else:
+            path = field["name"]
+        prop, dynamic = fill_section_properties(field, defaults, path)
+
+        # Only add properties if they have a content
+        if len(prop) is not 0:
+            properties[field.get("name")] = {"type": "nested", "properties": {}}
+            properties[field.get("name")]["properties"] = prop
+
+        dynamic_templates.extend(dynamic)
```

Comment:
I wonder if dynamic templates work on nested documents. We don't need it now, but we should know if that's a limitation.